File: frontend/lib/models/vehicle_geofence.dart
class VehicleGeofence {
  final int id;
  final int vehicleId;
  final int geofenceId;
  final DateTime? enteredAt;
  final DateTime? exitedAt;
  final bool isCurrentlyInside;
  final DateTime createdAt;

  VehicleGeofence({
    required this.id,
    required this.vehicleId,
    required this.geofenceId,
    this.enteredAt,
    this.exitedAt,
    required this.isCurrentlyInside,
    required this.createdAt,
  });

  factory VehicleGeofence.fromJson(Map<String, dynamic> json) {
    return VehicleGeofence(
      id: json['id'] as int,
      vehicleId: json['vehicleId'] as int,
      geofenceId: json['geofenceId'] as int,
      enteredAt: json['enteredAt'] != null
          ? DateTime.parse(json['enteredAt'] as String)
          : null,
      exitedAt: json['exitedAt'] != null
          ? DateTime.parse(json['exitedAt'] as String)
          : null,
      isCurrentlyInside: json['isCurrentlyInside'] as bool,
      createdAt: DateTime.parse(json['createdAt'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'vehicleId': vehicleId,
      'geofenceId': geofenceId,
      'enteredAt': enteredAt?.toIso8601String(),
      'exitedAt': exitedAt?.toIso8601String(),
      'isCurrentlyInside': isCurrentlyInside,
      'createdAt': createdAt.toIso8601String(),
    };
  }
}

File: dotnet-backend/projekatFlutter/Models/VehicleGeofence.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace projekatFlutter.Models
{
    public class VehicleGeofence:IEntity<int>
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int VehicleId { get; set; }

        [Required]
        public int GeofenceId { get; set; }

        public DateTime? EnteredAt { get; set; }
        public DateTime? ExitedAt { get; set; }

        [Required]
        public bool IsCurrentlyInside { get; set; } = false;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        
        [ForeignKey("VehicleId")]
        public Vehicle Vehicle { get; set; } = null!;

        [ForeignKey("GeofenceId")]
        public Geofence Geofence { get; set; } = null!;
    }
}